core/tests/sqlite/mod.rs

```diff
@@ -367,3 +368,153 @@ async fn test_sqlite_table_provider_roundtrip(
         "Round-tripped data should match original"
     );
 }
+
+#[tokio::test]
+async fn test_sqlite_decimal_memory() {
+    let arrow_record =
+        download_parquet_as_record_batch("https://public-data.spiceai.org/decimal.parquet")
```

Comment:
nitpick: It would be better not to depend on remote data and to include a test data file in the repository

Reply:
I view this as equivalent to how other projects do this (i.e. the datafusion-testing repo pulled in via submodule) - you don't really want to commit large files into git, so you need to pull it from somewhere.